src/stats/hexbin.jl

```diff
@@ -0,0 +1,136 @@
+"""
+    hexbin(xs, ys; kwargs...)
+
+Plots a heatmap with hexagonal bins for the observations `xs` and `ys`.
+
+## Attributes
+### Specific to `Hexbin`
+* `gridsize::Int = 20` sets the number of bins in x-direction
+* `mincnt::Int = 0` sets the minimal number of observations in the bin to be shown. If 0 all bins are shown, if 1 all with at least 1 observation.
+* `scale = identity` scales the number of data in the bins, eg. log10.
+### Generic
+* `colormap::Union{Symbol, Vector{<:Colorant}} = :viridis` sets the colormap that is sampled for numeric colors.
+* `colorrange::Tuple(<:Real,<:Real} = Makie.automatic`  sets the values representing the start and end points of `colormap`.
+"""
+@recipe(Hexbin) do scene
+    Attributes(;
+    colormap = theme(scene, :colormap),
+    colorrange=Makie.automatic,
+    gridsize = 20,
+    mincnt = 0,
+    scale=identity)
+end
+
+function Makie.plot!(hb::Hexbin{<:Tuple{<:AbstractVector{<:Any}, <:AbstractVector{<:Any}}})
+    x,y =hb[1:2]
+   
+    polys = Observable(Polygon[])
+    count_hex=Observable(Float64[])
+    function calculate_grid(x,y,grid_size,mincnt,scale)
+        empty!(polys[])
+        empty!(count_hex[])
+        if (length(x)==0)|(length(y)==0)
+            return
+        end
+        axis = current_axis()
```

Comment:
What exactly do you need the axis for? It looks like you're trying to access its limits? First of all, `current_axis()` isn't necessarily the axis you're currently plotting to, so you can't use that. Second, you're not supposed to access any axis stuff in a recipe, it lives a level below that in a simple Scene. (At least how recipes currently work). And third, you are only doing your calculation once, so if the limits changed the plot wouldn't update. You can access limits via the projection matrix of the Scene if you really need to (check out vlines etc for how it's done there) but it's kind of rare that we do this.